components/CreateNewPost/index.jsx

```diff
@@ -8,26 +8,45 @@ import { Headline } from "../ui/Headline/Headline.styled";
 import Form from "../Form";
 import { StyledFormSection } from "./CreateNewPost.styled";
 
-const CreateNewPost = () => {
+const CreateNewPost = ({ post, isEditing, onHandleIsEditing }) => {
+  const router = useRouter();
+  const [image, setImage] = useState("");
+
   const {
     register,
     handleSubmit,
     watch,
     formState: { errors },
-  } = useForm();
-  const router = useRouter();
-  const [image, setImage] = useState("");
+  } = useForm({
+    defaultValues: {
+      title: isEditing ? post.title : "",
+      description: isEditing ? post.description : "",
+      full_description: isEditing ? post.full_description : "",
+    },
+  });
 
   const onSubmit = async (fromData) => {
     const newObject = {
       image: "",
       ...fromData,
     };
-    const { data } = await axios.post(
-      process.env.NEXT_PUBLIC_DOMAIN + "/api/posts",
-      newObject
-    );
-    router.push(`/posts/${data._id}`);
+    try {
+      const { data } = isEditing
+        ? await axios.put(
+            process.env.NEXT_PUBLIC_DOMAIN + `/api/posts/${post._id}`,
+            newObject
+          )
+        : await axios.post(
+            process.env.NEXT_PUBLIC_DOMAIN + "/api/posts",
+            newObject
+          );
+
+      onHandleIsEditing && onHandleIsEditing();
```

Comment:
Why are you checking if `onHandleIsEditing` is true? You could just call the function since you're passing it as a prop anyway.
```suggestion
      onHandleIsEditing();
```